systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1066,6 +1070,301 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+                .withNewMetadata()
+                .withName(kafkaUserSource)
+                .endMetadata()
+                .addToData("password", "c291cmNlLXBhc3N3b3Jk")
+                .build();
+        kubeClient().patchSecret(namespaceName, kafkaUserSource, passwordSource);
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("Change KafkaUser sha-password on KMM2 Target");
+        Secret passwordTarget = new SecretBuilder()
+                .withNewMetadata()
+                .withName(kafkaUserTarget)
+                .endMetadata()
+                .addToData("password", "dGFyZ2V0LXBhc3N3b3Jk")
+                .build();
+        kubeClient().patchSecret(namespaceName, kafkaUserTarget, passwordTarget);
+        mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("Change CA cert for Source clusters by forcing annotation");
+        String clusterCaSecretName = KafkaResources.clusterCaCertificateSecretName(kafkaClusterSourceName);
+        Secret secret = new SecretBuilder(kubeClient(namespaceName).getSecret(namespaceName, clusterCaSecretName))
+            .editMetadata()
+                .addToAnnotations(Ca.ANNO_STRIMZI_IO_FORCE_RENEW, "true")
+            .endMetadata().build();
+        kubeClient(namespaceName).patchSecret(namespaceName, clusterCaSecretName, secret);
+        mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("Change CA cert for Target clusters by forcing annotation");
+        clusterCaSecretName = KafkaResources.clusterCaCertificateSecretName(kafkaClusterTargetName);
+        secret = new SecretBuilder(kubeClient(namespaceName).getSecret(namespaceName, clusterCaSecretName))
+            .editMetadata()
+                .addToAnnotations(Ca.ANNO_STRIMZI_IO_FORCE_RENEW, "true")
+            .endMetadata().build();
+        mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        kubeClient(namespaceName).patchSecret(namespaceName, clusterCaSecretName, secret);
+        DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        LOGGER.info("[TLS] Redeploy KMM2 with TLS authentication and change clients certificate");
+        KafkaMirrorMaker2ClusterSpec sourceClusterWithTlsAuth = new KafkaMirrorMaker2ClusterSpecBuilder()
```

Comment:
Can't you pass the already created spec to `KafkaMirrorMaker2ClusterSpecBuilder` and just change the needed fields?

Reply:
I haven't found the way on how to do it properly. There is no `toBuilder()` method on `KMM2CSpec`

Reply:
`new KafkaMirrorMaker2ClusterSpecBuilder(someOldSpec)` this isn't possible?

Reply:
Worked on that (with you). Threw that idea away as it was unnecessary complicated.